src/resmom/linux/alps.c

```diff
@@ -1003,7 +1001,14 @@ response_data_start(ud_t *d, const XML_Char *el, const XML_Char **atts)
 				brp->data.release.claims = 0;
 			} else if (strcmp(BASIL_VAL_QUERY, *vp) == 0) {
 				brp->method = basil_method_query;
-				brp->data.query.type = basil_query_none;
+				/*
+				 * Set type to status, for the switch status
+				 * response. The other types can get set in
+				 * inventory_start and engine_start.
+				 */
+				brp->data.query.type = basil_query_status;
```

Comment:
Hard to say how many times this code has changed over the years, but the original intent was to initialize the query type to none. It is later used to determine which (if any) memory needs to be freed in free_basil_resposne_data(). The handler for the engine query (engine_start) sets query type to basil_query_engine. I would expect the other handlers to do the same. Setting the query type to status could have unintended effect.

Reply:
@mike0042 Since there is no other way for us to identify that the query type as status, this seems to be the only viable option for now. I'd request you to please consider it again